assets/src/modules/SelectionTool.js

```diff
@@ -498,74 +413,87 @@ export default class SelectionTool {
-        if (geomOperator !== 'disjoint'){
-            getFeatureUrlData['options']['BBOX'] = selectionFeature.getGeometry().getExtent().join();
-            getFeatureUrlData['options']['SRSNAME'] = lConfig.crs;
+        // Restrict to current geometry extent for performance
+        // But not with 'disjoint' to get features
+        if (this._geomOperator !== 'disjoint') {
+            let geomExtent = geom.getExtent();
+            if (olExtent.getArea(geomExtent) == 0) {
+                geomExtent = olExtent.buffer(geomExtent, 0.000001);
+            }
+            wfsParams['BBOX'] = geomExtent;
+            wfsParams['SRSNAME'] = lConfig.crs;
         }
 
-        // get features
-        fetch(getFeatureUrlData['url'], {
-            method: "POST",
-            body: new URLSearchParams(getFeatureUrlData['options'])
-        }).then(response => {
-            return response.json();
-        }).then(result => {
-            const features = (new GeoJSON()).readFeatures(result, {
-                featureProjection: this._lizmap3.map.getProjection()
-            });
-
-            let sfIds = features.map(feat => feat.getId().split('.')[1]);
-
-            if (this._newAddRemoveSelected === 'add') {
-                sfIds = lConfig['selectedFeatures'].concat(sfIds);
-                for (let i = 0; i < sfIds.length; ++i) {
-                    for (let j = i + 1; j < sfIds.length; ++j) {
-                        if (sfIds[i] === sfIds[j])
-                            sfIds.splice(j--, 1);
-                    }
-                }
-            } else if (this._newAddRemoveSelected === 'remove') {
-                let asfIds = lConfig['selectedFeatures'].concat([]);
-                for (let i = 0; i < sfIds.length; ++i) {
-                    let asfIdIdx = asfIds.indexOf(sfIds[i]);
-                    if (asfIdIdx != -1)
-                        asfIds.splice(asfIdIdx, 1);
-                }
-                sfIds = asfIds;
+        wfs.getFeature(wfsParams).then(response => {
+            const features = (new GeoJSON()).readFeatures(response);
+
+            // Array of feature ids matching geometry condition
+            let featureIds = features.map(feature => feature.getId().split('.')[1]);
+
+            if (this.newAddRemoveSelected === 'add' ) { // Add to selection
+                featureIds = lConfig['selectedFeatures'].concat(featureIds);
+                // Remove duplicates
+                featureIds = [...new Set(featureIds)];
+            } else if (this.newAddRemoveSelected === 'remove' ) { // Remove from selection
+                const toRemove = new Set(featureIds);
+                featureIds = lConfig['selectedFeatures'].filter( x => !toRemove.has(x) );
             }
-            lConfig['selectedFeatures'] = sfIds;
+
+            lConfig['selectedFeatures'] = featureIds;
             lizMap.events.triggerEvent("layerSelectionChanged",
```

Comment:
```suggestion
            this._lizmap3.events.triggerEvent("layerSelectionChanged",
```
Do not use  `lizMap`